src/plugins/analysis/users_and_passwords/code/password_file_analyzer.py

```diff
@@ -6,113 +6,158 @@
 ]
 HTPASSWD_REGEXES = [
-    USER_NAME_REGEX + rb':\$apr1\$[a-zA-Z0-9\./+=]+\$[a-zA-Z0-9\./+]{22}',  # MD5 apr1
-    USER_NAME_REGEX + rb':\{SHA\}[a-zA-Z0-9\./+]{27}=',  # SHA-1
+    re.compile(USER_NAME_REGEX + rb':\$apr1\$[a-zA-Z0-9./+=]+\$[a-zA-Z0-9./+]{22}'),  # MD5 apr1
+    re.compile(USER_NAME_REGEX + rb':\{SHA}[a-zA-Z0-9./+]{27}='),  # SHA-1
 ]
-MOSQUITTO_REGEXES = [rb'[a-zA-Z][a-zA-Z0-9_-]{2,15}\:\$6\$[a-zA-Z0-9+/=]+\$[a-zA-Z0-9+/]{86}==']
+MOSQUITTO_REGEXES = [re.compile(rb'[a-zA-Z][a-zA-Z0-9_-]{2,15}:\$6\$[a-zA-Z0-9+/=]+\$[a-zA-Z0-9+/]{86}==')]
 RESULTS_DELIMITER = '=== Results: ==='
 
 
-class AnalysisPlugin(AnalysisBasePlugin):
-    """
-    This plug-in tries to find and crack passwords
-    """
-
-    NAME = 'users_and_passwords'
-    DEPENDENCIES = []  # noqa: RUF012
-    MIME_BLACKLIST = MIME_BLACKLIST_NON_EXECUTABLE
-    DESCRIPTION = 'search for UNIX, httpd, and mosquitto password files, parse them and try to crack the passwords'
-    VERSION = '0.5.4'
-    FILE = __file__
-
-    def process_object(self, file_object: FileObject) -> FileObject:
-        if self.NAME not in file_object.processed_analysis:
-            file_object.processed_analysis[self.NAME] = {}
-        file_object.processed_analysis[self.NAME]['summary'] = []
-        self.find_password_entries(file_object, UNIX_REGEXES, generate_unix_entry)
-        self.find_password_entries(file_object, HTPASSWD_REGEXES, generate_htpasswd_entry)
-        self.find_password_entries(file_object, MOSQUITTO_REGEXES, generate_mosquitto_entry)
-        return file_object
-
-    def find_password_entries(self, file_object: FileObject, regex_list: list[bytes], entry_gen_function: Callable):
-        for passwd_regex in regex_list:
-            passwd_entries = re.findall(passwd_regex, file_object.binary)
-            for entry in passwd_entries:
-                self.update_file_object(file_object, entry_gen_function(entry))
-
-    def _add_found_password_tag(self, file_object: FileObject, result: dict):
-        for password_entry in result:
-            if 'password' in result[password_entry]:
-                username = password_entry.split(':', 1)[0]
-                password = result[password_entry]['password']
-                self.add_analysis_tag(
-                    file_object, f'{username}_{password}', f'Password: {username}:{password}', TagColor.RED, True
-                )
-
-    def update_file_object(self, file_object: FileObject, result_entry: dict):
-        file_object.processed_analysis[self.NAME].update(result_entry)
-        file_object.processed_analysis[self.NAME]['summary'].extend(list(result_entry))
-        self._add_found_password_tag(file_object, result_entry)
-
-
-def generate_unix_entry(entry: bytes) -> dict:
+class CredentialResult(pydantic.BaseModel):
+    username: str = Field(description='The username.')
+    full_entry: str = Field(description='The full entry in unparsed form.')
```

Comment:
"full_entry" is a full, unparsed entry, e.g.

```
admin:$5$Mqs9JzinZfyjSMNR$M7DWqlMjKmOHwfzXnts89eQImN2QNDATeN/Z/BX0Jb.:12332:0:99999:7:::
```

or if it doesn't contain a pw hash:

```
admin:*:12332:0:99999:7:::
```

This can be interesting if something went wrong (or sees to have gone wrong) during parsing or if the result is empty (because there is no hash)